### PR TITLE
feat: ConfigProvider support Space classNames and styles properties

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
#42618
#42743

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     `ConfigProvider` support `Space` `classNames` and `styles` properties      |
| 🇨🇳 Chinese |       `ConfigProvider` 支持 `Space` 组件的 `classNames` 和 `styles` 属性     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e9a28d8</samp>

Enhanced the `Space` component to support customizing the `className` and `style` of its items through the `ConfigProvider` context. Added the `space` prop to the `ConfigProvider` component and updated the documentation and tests accordingly.

### <samp>🤖 Generated by Copilot at e9a28d8</samp>

*  Add `classNames` and `styles` props to the `ConfigProvider` component and context to allow customizing the `Space` component ([link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR72-R73), [link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R127-R128))
*  Import the `SpaceProps` type from `../space` in the `ConfigProvider` component and context files ([link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaR7), [link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R17))
*  Update the `ConfigProvider` documentation in English and Chinese to reflect the new `space` prop that can accept `classNames` and `styles` as well as `size` ([link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563L70-R70), [link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775L71-R71))
*  Modify the `Space` component to use the `classNames` and `styles` props from the `ConfigProvider` context if the `customClassNames` and `styles` props are not provided ([link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-d42c3072a5016964aad21722cb773b77e6fc5824adf56dc16efaecff82aab948L93-R96), [link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-d42c3072a5016964aad21722cb773b77e6fc5824adf56dc16efaecff82aab948L115-R118))
*  Import the `Space` component from `../../space` and add two test cases to check if the `ConfigProvider` component can pass custom `classNames` and `styles` props to the `Space` component ([link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5L9-R11), [link](https://github.com/ant-design/ant-design/pull/42748/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5R127-R166))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]

File: components/config-provider/context.ts [hunk 1]
```diff
@@ -68,6 +69,8 @@ export interface ConfigConsumerProps {
   direction?: DirectionType;
   space?: {
     size?: SizeType | number;
+    classNames?: SpaceProps['classNames'];
+    styles?: SpaceProps['styles'];
```
Comment (MadCcc) [inline components/config-provider/context.ts:73]: className 和 style 也加上吧

Comment (RedJue) [reply components/config-provider/context.ts:73]: ok

File: components/space/index.tsx [hunk 2]
```diff
@@ -150,6 +153,7 @@ const Space = React.forwardRef<HTMLDivElement, SpaceProps>((props, ref) => {
       ref={ref}
       className={cn}
       style={{
+        ...space?.style,
```
Comment (MadCcc) [inline components/space/index.tsx:156]: 这个放在下面一行好点

Comment (RedJue) [reply components/space/index.tsx:156]: 好，我调整下位置

**Pull request merged**